### ISSUE TITLE
custom error message causes typeError

### ISSUE THREAD

**Issue by yk1979** [colinhacks/zod#2155]
using zod v3.21.3

this code causes typeError, which is not wrong with previous version(v3.20.6).
```ts
const px =
  z.custom <
  `${number}px` >
  ((val) => /^\d+px$/.test(val as string), "custom error message");
```

error message
```
Argument of type 'string' is not assignable to parameter of type 'CustomParams | ((input: any) => CustomParams)'
```

here is my code sandbox link.
https://codesandbox.io/s/upbeat-williams-scfj81?file=/src/index.ts

**Comment by colinhacks** [comment]
Apologies, looking into this

**Comment by colinhacks** [comment]
This is fixed in Zod 3.21.4, but I'd recommend passing this value as a param called `message` instead:

```ts
const px = z.custom<`${number}px`>(
  (val) => /^\d+px$/.test(val as string),
  {message:"custom error message"}
);
```

**Comment by yk1979** [comment]
Thank you for your quick response!!!
